core/sdk-core/src/main/java/software/amazon/awssdk/core/exception/SdkException.java

```diff
@@ -133,17 +172,18 @@ public interface Builder extends Buildable {
     }
 
     protected static class BuilderImpl implements Builder {
-
         protected Throwable cause;
         protected String message;
+        protected Integer numAttempts = 0;
```

Comment:
Is there any reason we are defaulting to 0? That seems to defeat the purpose of `Integer`